### PR TITLE
Do not redirect, use render and reinitialize

### PR DESCRIPTION
**Pull request by ncounter**
Follow-up for https://github.com/openSUSE/open-build-service/pull/14080

### CODE REVIEW

File: src/api/app/controllers/webui/users/tokens_controller.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# rubocop:disable Metrics/ClassLength
```
Comment (danidoni) [inline src/api/app/controllers/webui/users/tokens_controller.rb:1]: I believe it's better to disable the cops with the [.rubocop_todo.yml](https://github.com/openSUSE/open-build-service/blob/9f99e2263a478c1d4a9a2d736f746ebdde61267c/.rubocop_todo.yml#L2)

File: src/api/app/controllers/webui/users/tokens_controller.rb [hunk 2]
```diff
@@ -105,19 +106,28 @@ def set_package
     # Check if only project_name or only package_name are present
     if @extra_params[:project_name].present? ^ @extra_params[:package_name].present?
       flash[:error] = 'When providing an optional package, both Project name and Package name must be provided.'
-      redirect_to new_token_path and return
+      @token = reinitialize_token(@params)
```
Comment (danidoni) [inline src/api/app/controllers/webui/users/tokens_controller.rb:109]: You can move this `@token` assignment and the one below in just one line above the line 105

File: src/api/app/controllers/webui/users/tokens_controller.rb [hunk 3]
```diff
@@ -105,19 +106,28 @@ def set_package
...
     rescue Package::UnknownObjectError
       flash[:error] = 'When providing an optional package, the package must exist. ' \
                       "Package '#{elide(@extra_params[:project_name])}/#{elide(@extra_params[:package_name])}' does not exist."
-      redirect_to new_token_path
+      @token = reinitialize_token(@params)
+      render :new
     end
   end
+
+  def reinitialize_token(params)
```
Comment (hennevogel) [inline src/api/app/controllers/webui/users/tokens_controller.rb:128]: I know what `Token.new` does, I have to look up what `reinitialize_token` does. I don't think it's worth turning this into a method...

**Review by hennevogel — CHANGES_REQUESTED** [review]

**Comment by ncounter** [comment]
Changes applied according to all the comments. Please could you have another look?

File: src/api/app/controllers/webui/users/tokens_controller.rb [hunk 4]
```diff
@@ -102,22 +102,25 @@ def set_package
     # Prevent setting a package for a workflow token
     return if @params[:type] == 'workflow'
 
+    # Make sure token in initialized before running into initialization errors and return
```
[the remark this answers is not in the document]
Comment (ncounter) [reply src/api/app/controllers/webui/users/tokens_controller.rb]: I rephrased a little bit. The point is making sense of the _why are we initializing the variable `@token` here?_

[the remark this answers is not in the document]
Comment (hennevogel) [reply src/api/app/controllers/webui/users/tokens_controller.rb]: I don't think you clarify anything with this. Settings objects up before the controller action is standard practice. Usually even in a `before_action`....

[the remark this answers is not in the document]
Comment (ncounter) [reply src/api/app/controllers/webui/users/tokens_controller.rb]: I see. I'll just drop the comment then.

File: src/api/.rubocop_todo.yml [hunk 5]
```diff
@@ -168,6 +168,7 @@ Metrics/ClassLength:
     - 'app/controllers/webui/search_controller.rb'
     - 'app/controllers/webui/staging/workflows_controller.rb'
     - 'app/controllers/webui/users_controller.rb'
+    - 'app/controllers/webui/users/tokens_controller.rb'
```
Comment (hennevogel) [inline src/api/.rubocop_todo.yml:171]: This is needed because of one extra line?

Comment (ncounter) [reply src/api/.rubocop_todo.yml]: Apparently not. It was before with few more lines, but now it doesn't. Good catch. Dropped

**Review by hennevogel — APPROVED** [review]

**Pull request merged**